### ISSUE TITLE
How to draw the fig???

### ISSUE THREAD

**Issue by miaozhang0525** [xiangning-chen/SmoothDARTS#2]
Hi, Xiangning,

I am very interested in your paper and your code. Great job!
 My concern is how to draw the figure1 in your paper.

Look forward to your explanation.

**Comment by xiangning-chen** [comment]
Thanks for your attention, the X-axis is the gradient direction and the Y-axis is another random orthogonal direction. Every data point (x,y) represents the validation accuracy achieved by A+x*gradient+y*orthogonal, meanwhile the weights W are kept the same.

**Comment by Jialiang14** [comment]
Hi, Xiangning,

Your job is excellent ! I am also very interested in the figure 1. (x, y) is a point, however, the gradient of validation loss on A is a vector. Maybe you assume that each element in A is set to the same. I am sorry for that I still could understand the drawing of figure 1 difficultly. 

I have read your work and the paper "Understanding and Robustifying Differentiable Architecture Search" together.  The only confusion is about the figure 1.

Look forward to your reply.

Best wishes
Jialiang Sun

**Comment by xiangning-chen** [comment]
Hi,

Thanks for your attention. The (x,y) is just a scale rather than gradient, you can imagine that it's alike learning rate.
As I mentioned before, every data point is the validation accuracy of the super-network with architecture weight "A+x * gradient+y * orthogonal".

Thanks

**Comment by GongXinyuu** [comment]
> Hi,
> 
> Thanks for your attention. The (x,y) is just a scale rather than gradient, you can imagine that it's alike learning rate.
> As I mentioned before, every data point is the validation accuracy of the super-network with architecture weight "A+x * gradient+y * orthogonal".
> 
> Thanks

Thanks for your clarification. Just one more question. Where does the gradient here come from? Does it come from an arbitrary arch param BP during searching?

**Comment by xiangning-chen** [comment]
Hi,

Thanks for your question. 
The gradient is derived from the trained supernet when the search is finished.

**Comment by GongXinyuu** [comment]
@xiangning-chen Thanks for your prompt reply!
